commands/templates/addons/env/opensearch.yml

```diff
@@ -152,7 +152,7 @@ Resources:
             AWS: '*'
           Action:
           - 'es:ESHttp*'
-          Resource: !Sub 'arn:aws:es:${AWS::Region}:${AWS::AccountId}:domain/${App}-${Env}-{{ service.name }}/*'
+          Resource: !Sub 'arn:aws:es:${AWS::Region}:${AWS::AccountId}:domain/{{ service.name|replace("-", "") }}*'
```

Comment:
Will this cause problems if you have two environments in an account, e.g dev and staging, and they each have "mydatabase"?

Reply:
This is down to the way the opensearch cloudformation resource works. The opensearch domain if named [cannot be updated in a way that would replace the service](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-opensearchservice-domain.html#cfn-opensearchservice-domain-domainname) and the default arn convention is `arn:aws:es:[region]:[account]:domain/[logical id without hyphens max 16 chars]-[random string]`.

This access policy will make the claim (that will be ignored) that anyone can access the other opensearch domains on the account, but as the principal is any resource in the account this is the case anyway.